Adds GitHub Actions CI for pipelines

swiftdiaries opened the pull request:
Currently, the GitHub Action uses a managed KF Pipelines Host and a ubuntu-runner.
In the future, this could be a self-hosted runner on UCS.

What the workflow does?
- Runs the "CI" version of the pipeline notebook using [Papermill](https://github.com/nteract/papermill)
- Creates a deployable artifact for the pipeline (Argo YAML)
- Connects to the Managed pipeline using Secrets set in the repo
  - One concern; all collaborators who submit a PR can access the KFP instance

amsaha commented:
This is a good start. Two questions:

1. I suppose this is running right now on GKE. Where are the account credentials stored? 
2. Right now it seems that the CI test failed. Does that need to be fixed?

swiftdiaries commented:
> Thanks @swiftdiaries. This is a good start. A few questions:
> 
>     1. I suppose this is running right now on GKE. Where are the account credentials stored?

The secrets are stored in the repo. [GitHub Secrets Documentation.](https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets)

>     2. Currently, what is exactly being checked here (in the "Run and Check Pipeline Output")? Does not have to be anything at this point (since we are just starting off).

It runs the pipeline and checks if it was successfully triggered.
I'm having a bit of trouble invoking secrets in workflows initiated by PRs from forked repo's. Like this PR.
Right now, I've moved the logic for checking into the Generate step and for now it checks only if the pipeline conforms to a valid Argo workflow.

>     3. Right now it seems that the CI test failed. Does that need to be fixed?

Fixed it by moving the checking logic into the Generate pipeline step. And removing the access to GCP KF pipelines service

swiftdiaries commented:
This is ready for another review @amsaha
